From 52f9cda2fd46d1247a08b675503ea9862fb999d3 Mon Sep 17 00:00:00 2001
From: Prokhar Bhowal <Prokhar.Bhowal@amd.com>
Date: Thu, 18 Aug 2016 12:13:54 -0400
Subject: [PATCH 0844/1722] drm/amd/dal: Static Screen

- only enable static screen on embedded panels
(not on external monitors)

Signed-off-by: Prokhar Bhowal <Prokhar.Bhowal@amd.com>
Acked-by: Harry Wentland <harry.wentland@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
Signed-off-by: Kalyan Alle <kalyan.alle@amd.com>
---
 drivers/gpu/drm/amd/dal/modules/freesync/freesync.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/amd/dal/modules/freesync/freesync.c b/drivers/gpu/drm/amd/dal/modules/freesync/freesync.c
index 1e78209..88485b2 100644
--- a/drivers/gpu/drm/amd/dal/modules/freesync/freesync.c
+++ b/drivers/gpu/drm/amd/dal/modules/freesync/freesync.c
@@ -780,8 +780,14 @@ void mod_freesync_update_state(struct mod_freesync *mod_freesync,
 			freesync_program_required = true;
 			break;
 		case FREESYNC_STATE_STATIC_SCREEN:
-			/* Change core variables only if there is a change*/
-			if (state->static_screen != freesync_params->enable) {
+			/* Static screen ramp is only enabled for embedded
+			 * panels. Also change core variables only if there
+			 * is a change.
+			 */
+			if (dc_is_embedded_signal(
+				streams[stream_index]->sink->sink_signal) &&
+				state->static_screen !=
+				freesync_params->enable) {
 
 				/* Change the state flag */
 				state->static_screen = freesync_params->enable;
-- 
2.7.4

